src/java.security.jgss/windows/native/libsspi_bridge/sspi.cpp

```diff
@@ -368,54 +368,58 @@ gss_import_name(OM_uint32 *minor_status,
     }
 
     SEC_WCHAR* value = new SEC_WCHAR[len + 1];
-    if (value == NULL) {
-        goto err;
-    }
-
-    len = MultiByteToWideChar(CP_UTF8, 0, input, len, value, len+1);
-    if (len == 0) {
-        goto err;
-    }
-    value[len] = 0;
 
-    PP("import_name from %ls", value);
+    {
```

Comment:
I agree that it's ugly, but at the time I couldn't think of another way to solve the issue. By any chance, why does splitting it out into separate declaration assignment work? Last I remember, it still jumps over the local even when split up like this.

Reply:
Well, SO offers a speculative answer:
https://stackoverflow.com/a/14274292
the modified form is not more safe, but the standard does not forbid it.

Reply:
I tested the new approach on gcc, it worked, so I'll commit it here

Reply:
I've changed it to use the split approach, but this feels like a hack to me. I wonder if there's a better way of doing this?

Reply:
well goto was always a hack; less hacky solution would involve splitting the giant function into smaller parts, so that replacing goto with an else block doesn't increase indentation too much. But if it ain't broke...